web/api/web_api_v1.c

```diff
@@ -1191,6 +1191,50 @@ int web_client_api_request_v1_anomaly_event_info(RRDHOST *host, struct web_clien
     freez(s);
     return HTTP_RESP_OK;
 }
+
+int web_client_api_request_v1_anomaly_rate_info(RRDHOST *host, struct web_client *w, char *url) {
+    if (!netdata_ready)
+        return HTTP_RESP_BACKEND_FETCH_FAILED;
+
+    uint32_t after = 0, before = 0;
+
+    while (url) {
+        char *value = mystrsep(&url, "&");
+        if (!value || !*value)
+            continue;
+
+        char *name = mystrsep(&value, "=");
+        if (!name || !*name)
+            continue;
+        if (!value || !*value)
+            continue;
+
+        if (!strcmp(name, "after"))
+            after = (uint32_t) (strtoul(value, NULL, 0) / 1000);
+        else if (!strcmp(name, "before"))
+            before = (uint32_t) (strtoul(value, NULL, 0) / 1000);
+    }
```

Comment:
@siamaktavakoli The timestamps in the URL are in milliseconds. The ML code works with timestamps that are in seconds. The conversion from milliseconds to seconds happens in these lines.

Reply:
noted.